### PR TITLE
Jmu/elastic additions

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

File: src/main/java/sirius/db/es/ElasticEntity.java [hunk 1]
```diff
@@ -64,4 +71,39 @@ public String getId() {
...
+     *
+     * @param queryName the name of the query to check
+     * @return the list of named queries which matched this entity.
+     */
+    public boolean isMatchedNamedQuery(String queryName) {
+        if (matchedQueries == null) {
+            return false;
+        }
+
+        return matchedQueries.stream().anyMatch(query -> Strings.areEqual(query, queryName));
```
Comment (andyHa) [inline src/main/java/sirius/db/es/ElasticEntity.java]: make matchedQueries a set?

File: src/main/java/sirius/db/es/ElasticEntity.java [hunk 2]
```diff
@@ -64,4 +71,39 @@ public String getId() {
     protected void setId(String id) {
         this.id = id;
     }
+
+    /**
+     * Gets the list of all named queries which matched this entity.
```
Comment (andyHa) [inline src/main/java/sirius/db/es/ElasticEntity.java]: add a note that this is only populated if the entity is the result of a query

File: src/main/java/sirius/db/mixing/properties/AmountProperty.java [hunk 3]
```diff
@@ -87,12 +90,17 @@ public Object transformValue(Value value) {
     }
 
     @Override
-    protected Object transformToDatasource(Class<? extends BaseMapper<?, ?, ?>> mapperType, Object object) {
+    protected Object transformToJDBC(Object object) {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/AmountProperty.java]: what about mongo DB - no love?

File: src/main/java/sirius/db/mixing/properties/AmountProperty.java [hunk 4]
```diff
@@ -87,12 +90,17 @@ public Object transformValue(Value value) {
...
         return object == null || ((Amount) object).isEmpty() ? null : ((Amount) object).getAmount();
     }
 
     @Override
-    protected Object transformFromDatasource(Class<? extends BaseMapper<?, ?, ?>> mapperType, Value data) {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/AmountProperty.java]: s.a.

File: src/main/java/sirius/db/es/ElasticQuery.java [hunk 5]
```diff
@@ -174,6 +177,18 @@ public ElasticQuery(EntityDescriptor descriptor, LowLevelClient client) {
...
 
+    /**
+     * Enables the explain mode which gives detailed informations about score calculations.
+     * <p>
+     * Only use this mode for debugging as this might cost performance!
+     *
+     * @return the query itself for fluent method calls
+     */
+    public ElasticQuery<E> explain() {
+        explain = true;
```
Comment (andyHa) [inline src/main/java/sirius/db/es/ElasticQuery.java]: this.explain

Comment (andyHa) [reply src/main/java/sirius/db/es/ElasticQuery.java]: as it is sorta kinda a setter

File: src/main/java/sirius/db/mixing/properties/AmountProperty.java (hunk 3 above)
Comment (jmuscireum) [reply src/main/java/sirius/db/mixing/properties/AmountProperty.java]: This should be a separate PR: SE-5038

File: src/main/java/sirius/db/mixing/properties/AmountProperty.java (hunk 4 above)
Comment (jmuscireum) [reply src/main/java/sirius/db/mixing/properties/AmountProperty.java]: dito

**Pull request merged**